src/openapi-metadata.ts

```diff
@@ -0,0 +1,69 @@
+import { SchemaObject, ParameterObject, ReferenceObject } from 'openapi3-ts';
+import { ZodSchema } from 'zod';
+import { omitBy, isNil, omit } from './lib/lodash';
+import { isZodType } from './lib/zod-is-type';
+import { ZodOpenAPIMetadata } from './zod-extensions';
+
+export class OpenAPIMetadata {
+  static buildParameterMetadata(
+    metadata: Required<ZodOpenAPIMetadata>['param']
+  ) {
+    return omitBy(metadata, isNil);
+  }
+
+  static getMetadata<T extends any>(
+    zodSchema: ZodSchema<T>
+  ): ZodOpenAPIMetadata<T> | undefined {
+    const innerSchema = this.unwrapChained(zodSchema);
+    const metadata = zodSchema._def.openapi
+      ? zodSchema._def.openapi
+      : innerSchema._def.openapi;
+
+    return metadata;
+  }
+
+  static applySchemaMetadata(
```

Comment:
I'm not a fan of the statics, can you please make it a normal object? We can instantiate it once in `OpenApiGenerator`.